File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n=height.size();
        int i=0;
        int j=n-1;
        int area=0;
        while(i<=j){
            int h=min(height[i],height[j]);
            int w=j-i;
            area=max(area,h*w);
            if(height[i]<=height[j]) i++;
            else j--;
        }
        return area;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n=height.length;

        int i=0;
        int j=n-1;

        int ans=0;
        while(i<j){
            int heights=Math.min(height[i],height[j]);
            int width=j-i;

            ans=Math.max(heights*width,ans);
            if(height[i]<height[j]) i++;
            else j--;
        }
        return ans;
    }
}